apps/designer/app/designer/features/sidebar-left/panels/asset-manager.tsx

```diff
@@ -2,48 +2,53 @@ import { ImageIcon } from "~/shared/icons";
 import { Flex, Grid, Heading } from "~/shared/design-system";
 import { useEffect, useState } from "react";
 import { useActionData } from "@remix-run/react";
-import { Asset } from "@webstudio-is/prisma-client";
 import { AssetManagerImage } from "./components/image";
 
 import { AddAnAssetForm } from "./components/add-an-asset-form";
-import { UploadingAsset } from "../types";
+import { Asset } from "../types";
 
-export const TabContent = ({
-  assets: baseAssets,
-}: {
-  assets: Array<Asset | UploadingAsset>;
-}) => {
-  const newImages = useActionData();
+export const useAssetsState = (baseAssets: Array<Asset>) => {
+  const imageChanges = useActionData();
 
-  const [assets, setAssets] = useState(baseAssets);
+  const [assets, setAssets] = useState<Asset[]>(baseAssets);
 
   useEffect(() => {
-    if (newImages?.length) {
+    if (imageChanges?.uploadedAssets?.length) {
       setAssets((currentAssets) => [
-        ...newImages,
+        ...imageChanges.uploadedAssets,
         ...currentAssets.filter((asset) => asset.status !== "uploading"),
       ]);
     }
-  }, [newImages]);
+    if (imageChanges?.deletedAsset?.id) {
+      setAssets((currentAssets) => [
+        ...currentAssets.filter(
+          (asset) => asset.id !== imageChanges.deletedAsset.id
+        ),
+      ]);
+    }
+  }, [imageChanges]);
 
+  const onUploadAsset = (uploadedAssets: Array<Asset>) =>
+    setAssets((assets) => [...uploadedAssets, ...assets]);
+
+  return { assets, onUploadAsset };
+};
+
+export const TabContent = ({
+  assets: baseAssets,
+}: {
+  assets: Array<Asset>;
+}) => {
+  const { assets, onUploadAsset } = useAssetsState(baseAssets);
   return (
     <Flex gap="3" direction="column" css={{ padding: "$1" }}>
-      <Flex justify="between">
+      <Flex justify="between" align="center">
         <Heading>Assets</Heading>
-        <AddAnAssetForm
-          onSubmit={(uploadedAssets: Array<UploadingAsset>) =>
-            setAssets((assets) => [...uploadedAssets, ...assets])
-          }
-        />
+        <AddAnAssetForm onSubmit={onUploadAsset} />
       </Flex>
       <Grid columns={2} gap={2}>
         {assets.map((asset) => (
-          <AssetManagerImage
-            key={asset.id}
-            path={asset.path}
-            alt={asset.alt || asset.name}
-            status={asset.status}
-          />
+          <AssetManagerImage key={asset.id} {...asset} />
```

Comment:
This component is so much better off with a "Thumbnail" as a name :)

Reply:
updated!